src/read/coff/import.rs

```diff
@@ -0,0 +1,179 @@
+//! Support for reading short import files.
+//!
+//! These are used by some Windows linkers as a more compact way to describe
+//! dynamically imported symbols.
+
+use crate::read::{Architecture, Error, ReadError, ReadRef, Result};
+use crate::{pe, ByteString, Bytes, LittleEndian as LE};
+
+/// A Windows short form description of a symbol to import.
+/// Used in Windows import libraries. This is not an object file.
+#[derive(Debug, Clone)]
+pub struct MsImportFile<'data> {
+    header: &'data pe::ImportObjectHeader,
+    dll: ByteString<'data>,
+    symbol: ByteString<'data>,
+    kind: ImportType,
+    import: Option<ByteString<'data>>,
+}
+impl<'data> MsImportFile<'data> {
+    /// Parse it.
+    pub fn parse<R: ReadRef<'data>>(data: R) -> Result<Self> {
+        let mut offset = 0;
+        let header = pe::ImportObjectHeader::parse(data, &mut offset)?;
+        let data_size = header.size_of_data.get(LE);
+        let mut strings = Bytes(
+            data.read_bytes(&mut offset, data_size as u64)
+                .read_error("Invalid COFF import library data size")?,
+        );
+        let symbol = strings
+            .read_string()
+            .read_error("Could not read COFF import library symbol name")?;
+        let dll = strings
+            .read_string()
+            .read_error("Could not read COFF import library DLL name")?;
+
+        // Unmangles a name by removing a `?`, `@` or `_` prefix.
+        fn strip_prefix(s: &[u8]) -> &[u8] {
+            match s.split_first() {
+                Some((b'?' | b'@' | b'_', rest)) => rest,
+                _ => s,
+            }
+        }
+        Ok(Self {
+            header,
+            dll: ByteString(dll),
+            symbol: ByteString(symbol),
+            kind: ImportType::from_u2(header.name_type.get(LE) & 0b11)?,
+            import: match ImportNameType::from_u3((header.name_type.get(LE) >> 2) & 0b111)? {
```

Comment:
Converting to `ImportNameType` seems like an unnecessary step, since `ImportNameType` is never used anywhere other than here. We could directly match on the `pe` constants instead.